pkg/api/v1beta3/types.go

```diff
@@ -509,6 +509,33 @@ type EnvVar struct {
 	Name string `json:"name" description:"name of the environment variable; must be a C_IDENTIFIER"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty" description:"value of the environment variable; defaults to empty string"`
+	// Optional: specify a source for the value of the EnvVar.
+	*EnvVarSource `json:"source,omitempty" description:"source for the environment variable's value; cannot be used if value is set"`
```

Comment:
If we think there may be multiple sources in the future, then I prefer the extra level of nesting. I was objecting to the specific combination of a pointer to an anonymous Go struct. Name the field Source.